client/state/plugins/installed/test/selectors-ts.ts

```diff
@@ -213,3 +229,108 @@ describe( 'getAllPluginsIndexedBySiteId', () => {
+		const siteIds = [ nonExistingSiteId1, nonExistingSiteId2 ];
+		expect( getPluginOnSites( state, siteIds, 'akismet' ) ).toBeUndefined();
+	} );
+
+	test( 'Should get an undefined value if the requested plugin on these sites is not in the current state', () => {
+		expect( getPluginOnSites( state, [ siteOneId ], 'jetpack' ) ).toBeUndefined();
+	} );
+
+	test( 'Should get the plugin if the it exists on one or more of the requested sites', () => {
+		const siteIds = [ siteOneId, siteTwoId ];
+		const plugin = getPluginOnSites( state, siteIds, 'hello-dolly' );
+		const sitesWithPlugins = {
+			[ siteOneId ]: {
+				active: helloDolly.active,
+				autoupdate: helloDolly.autoupdate,
+				update: helloDolly.update,
+				version: helloDolly.version,
+			},
+			[ siteTwoId ]: {
+				active: helloDolly.active,
+				autoupdate: helloDolly.autoupdate,
+				update: helloDolly.update,
+				version: helloDolly.version,
+			},
+		};
+		const { active, autoupdate, update, version, ...pluginProperties } = helloDolly;
+		expect( plugin ).toEqual( { ...pluginProperties, sites: sitesWithPlugins } );
+	} );
+} );
+
+describe( 'getPluginOnSite', () => {
+	test( 'Should get an undefined value if the requested site is not in the current state', () => {
+		expect( getPluginOnSite( state, nonExistingSiteId1, 'akismet' ) ).toBeUndefined();
+	} );
+
+	test( 'Should get an undefined value if the requested plugin on this site is not in the current state', () => {
+		expect( getPluginOnSite( state, siteOneId, 'jetpack' ) ).toBeUndefined();
+	} );
+
+	test( 'Should get the plugin if the it exists on the requested site', () => {
+		const plugin = getPluginOnSite( state, siteOneId, 'akismet' );
+		const siteProperties = {
+			active: akismet.active,
+			autoupdate: akismet.autoupdate,
+			update: akismet.update,
+			version: akismet.version,
+		};
+		const expectedPlugin = {
+			...akismet,
+			...{ sites: { [ siteOneId ]: siteProperties } },
+		};
+		expect( plugin ).toEqual( expectedPlugin );
+	} );
+} );
+
+describe( 'getPuginsOnSite', () => {
+	test( 'Should get an array of undefined if the requested site is not in the current state', () => {
+		expect( getPluginsOnSite( state, siteThreeId, [ 'jetpack' ] ) ).toEqual( [ undefined ] );
+	} );
+
+	test( 'Should get an  array of undefined if the requested site has no plugins', () => {
+		expect( getPluginsOnSite( state, siteFourId, [ 'akismet' ] ) ).toEqual( [ undefined ] );
+	} );
+
+	test( 'Should get an array with the plugin objects for the requested plugins with site properties lifted onto the plugin object', () => {
+		const plugins = getPluginsOnSite( state, siteOneId, [ 'akismet' ] );
+		expect( plugins ).toEqual( [
+			{
+				...akismetWithSites,
+				...{
+					active: akismetWithSites.sites[ siteOneId ].active,
+					autoupdate: akismetWithSites.sites[ siteOneId ].autoupdate,
+					version: akismetWithSites.sites[ siteOneId ].version,
```

Comment:
Intentionally missing the `update` key?

Reply:
Yes, there is no `update` object on the akismet plugin fixture which simulates the case where a plugin has no updates.